ksql-engine/src/test/java/io/confluent/ksql/function/udf/math/RoundTest.java

```diff
@@ -0,0 +1,232 @@
+    assertThat(udf.round(1.53e6d), is(1530000L));
+    assertThat(udf.round(10.01d), is(10L));
+    assertThat(udf.round(12345.5d), is(12346L));
+    assertThat(udf.round(9.99d), is(10L));
+    assertThat(udf.round(110.1), is(110L));
+    assertThat(udf.round(1530000.01d), is(1530000L));
+    assertThat(udf.round(9999999.99d), is(10000000L));
+  }
+
+  @Test
+  public void testRoundSimpleDoubleNegative() {
+    assertThat(udf.round(-1.23d), is(-1L));
+    assertThat(udf.round(-1.0d), is(-1L));
+    assertThat(udf.round(-1.5d), is(-1L));
+    assertThat(udf.round(-1.75d), is(-2L));
+    assertThat(udf.round(-1.53e6d), is(-1530000L));
+    assertThat(udf.round(-10.01d), is(-10L));
+    assertThat(udf.round(-12345.5d), is(-12345L));
+    assertThat(udf.round(-9.99d), is(-10L));
+    assertThat(udf.round(-110.1), is(-110L));
+    assertThat(udf.round(-1530000.01d), is(-1530000L));
+    assertThat(udf.round(-9999999.99d), is(-10000000L));
+  }
+
+  @Test
+  public void testRoundSimpleBigDecimalPositive() {
+    assertThat(udf.round(new BigDecimal("0.0")), is(new BigDecimal("0")));
+    assertThat(udf.round(new BigDecimal("1.23")), is(new BigDecimal("1")));
+    assertThat(udf.round(new BigDecimal("1.0")), is(new BigDecimal("1")));
+    assertThat(udf.round(new BigDecimal("1.5")), is(new BigDecimal("2")));
+    assertThat(udf.round(new BigDecimal("1.75")), is(new BigDecimal("2")));
+    assertThat(udf.round(new BigDecimal("1530000")), is(new BigDecimal("1530000")));
+    assertThat(udf.round(new BigDecimal("10.1")), is(new BigDecimal("10")));
+    assertThat(udf.round(new BigDecimal("12345.5")), is(new BigDecimal("12346")));
+    assertThat(udf.round(new BigDecimal("9.99")), is(new BigDecimal("10")));
+    assertThat(udf.round(new BigDecimal("110.1")), is(new BigDecimal("110")));
+    assertThat(udf.round(new BigDecimal("1530000.01")), is(new BigDecimal("1530000")));
+    assertThat(udf.round(new BigDecimal("9999999.99")), is(new BigDecimal("10000000")));
+  }
+
+  @Test
+  public void testRoundSimpleBigDecimalNegative() {
+    assertThat(udf.round(new BigDecimal("-1.23")), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1.0")), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1.5")), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1530000")), is(new BigDecimal("-1530000")));
+    assertThat(udf.round(new BigDecimal("-10.1")), is(new BigDecimal("-10")));
+    assertThat(udf.round(new BigDecimal("-12345.5")), is(new BigDecimal("-12345")));
+    assertThat(udf.round(new BigDecimal("-9.99")), is(new BigDecimal("-10")));
+    assertThat(udf.round(new BigDecimal("-110.1")), is(new BigDecimal("-110")));
+    assertThat(udf.round(new BigDecimal("-1530000.01")), is(new BigDecimal("-1530000")));
+    assertThat(udf.round(new BigDecimal("-9999999.99")), is(new BigDecimal("-10000000")));
+  }
+
+  @Test
+  public void testRoundDoubleWithDecimalPlacesPositive() {
```

Comment:
nit: test name is 'xxxWithDecimalPlacesPositive`, but includes tests with negative scale...

Reply:
The negative is referring to the value, not the scale